crates/sui-indexer-alt-graphql/src/task/watermark.rs

```diff
@@ -221,10 +221,12 @@ impl Watermarks {
     }
 
     /// The reader_lo for a pipeline. Returned as an inclusive checkpoint number.
-    pub(crate) fn pipeline_lo_watermark(&self, pipeline: &str) -> anyhow::Result<&Watermark> {
+    pub(crate) fn pipeline_lo_watermark(&self, pipeline: &str) -> Result<&Watermark, RpcError> {
         self.pipeline_lo
             .get(pipeline)
-            .ok_or_else(|| anyhow!("'{pipeline}' not found in pipeline_lo watermarks"))
+            .ok_or_else(|| RpcError::FeatureUnavailable {
+                what: "pipeline data",
```

Comment:
The error here needs a bit more nuance -- if you hit this case while trying to paginate, then the issue is not that you're missing the pipeline data, it means that you're missing whatever feature the pipeline represents (e.g. missing `tx_affected_addresses` means you can't paginate transaction by their affected addresses, etc).

Also, if you hit this case when you are calling `AvailableRange` then should that result in an error too, or should that result in returning an empty available range?

Reply:
Maybe the right place to format that error message is around `AvailableRangeKey`, where you have the type, field, and filters around.